### PR TITLE
Add timeout for process WaitForExit

### PR DESCRIPTION
**Pull request by smadala**
- Vstest.console waiting indefinitely if testhost launch a process and won't kill it.

### CODE REVIEW

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs [hunk 1]
```diff
@@ -48,11 +48,12 @@ public object LaunchProcess(string processPath, string arguments, string working
...
                     process.Exited += (sender, args) =>
                     {
-                        // Call WaitForExit again to ensure all streams are flushed
+                        // Call WaitForExit without again to ensure all streams are flushed,
+                        // Add timeout to avoid indefinate waiting on child process exists.
                         var p = sender as Process;
                         try
                         {
-                            p.WaitForExit();
+                            p.WaitForExit(500);
```
Comment (codito) [inline src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs]: Timeouts are approximate. What is the worst case? Is it only limited to loss of error stream data?

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs [hunk 2]
```diff
@@ -48,11 +48,12 @@ public object LaunchProcess(string processPath, string arguments, string working
                 {
                     process.Exited += (sender, args) =>
                     {
-                        // Call WaitForExit again to ensure all streams are flushed
+                        // Call WaitForExit without again to ensure all streams are flushed,
+                        // Add timeout to avoid indefinate waiting on child process exists.
```
Comment (codito) [inline src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs]: nit: s/indefinate/indefinite

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs (hunk 1 above)
Comment (navin22) [reply src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs]: Is it good to take this timeout as part of argument while launching the process ?

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs (hunk 2 above)
Comment (smadala) [reply src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs]: Done.

File: src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs (hunk 1 above)
Comment (smadala) [reply src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs]: @codito Verified writing 1.3MB of data continuously to testhost standard error, which got captured in vstest.console within 500 ms.  Yes, Theoretically error stream can be lost. At cost of losing error stream we can avoid hang.
@navin22 As scenarios is very rare, I think adding this variable part of arguments won't required.

**Review by codito — APPROVED** [review]

**Pull request merged**